### PR TITLE
feat: add aggregate count functions with decimal return type

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: extensions/functions_aggregate_decimal.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+%YAML 1.2
+---
+aggregate_functions:
+  - name: "count"
```
Comment (Blizzara) [inline extensions/functions_aggregate_decimal.yaml:4]: overloading functions based only on return type causes problems, see e.g. https://github.com/substrait-io/substrait/pull/606#issuecomment-1964054841

Comment (scgkiran) [reply extensions/functions_aggregate_decimal.yaml:4]: I have them in a different yaml file. can this still cause problems?

Comment (jacques-n) [reply extensions/functions_aggregate_decimal.yaml:4]: different files should be ok.

Comment (Blizzara) [reply extensions/functions_aggregate_decimal.yaml:4]: I believe it'll still break substrait-java, see https://github.com/substrait-io/substrait-java/issues/275

Comment (jacques-n) [reply extensions/functions_aggregate_decimal.yaml:4]: I think we should fix that in substrait-java then as opposed to holding back the specification. Extensions being independent is key to extensions being extensions :)

File: extensions/functions_aggregate_decimal.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+%YAML 1.2
+---
+aggregate_functions:
```
Comment (jacques-n) [inline extensions/functions_aggregate_decimal.yaml:3]: the name of this file is weird. in other cases type names in filenames are about input types and this one is about output types. I suggest we rename file to something that is more descriptive. Maybe also add a little text to the functions clarifying how they are different than classic count functions.

**Review by jacques-n — CHANGES_REQUESTED** [review]

File: extensions/functions_aggregate_decimal.yaml (hunk 1 above)
Comment (vbarua) [reply extensions/functions_aggregate_decimal.yaml]: It's possible for us to work around duplicates in substrait-java by excluding them from the default extensions for now. See https://github.com/substrait-io/substrait-java/issues/275#issuecomment-2257176686

**Review by vbarua — CHANGES_REQUESTED** [review]
Overall these changes look reasonable to me. Just one suggestion about the name `count_star`.

File: extensions/functions_aggregate_decimal_output.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          - name: x
+            value: any
+        options:
+          overflow:
+            values: [SILENT, SATURATE, ERROR]
+        nullability: DECLARED_OUTPUT
+        decomposable: MANY
+        intermediate: decimal<38,0>
+        return: decimal<38,0>
+  - name: "count_star"
```
[the remark this answers is not in the document]
Comment (scgkiran) [reply extensions/functions_aggregate_decimal_output.yaml]: changed it count

**Review by vbarua — APPROVED** [review]
Changes look good to me.

**Review by jacques-n — APPROVED** [review]
lgtm

**Pull request merged**